workflow-search/templates/deployment.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "workflow-search.fullname" . }}
+  labels:
+{{ include "workflow-search.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "workflow-search.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "workflow-search.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /actuator/health
+              port: {{ .Values.appConfig.port }}
+            initialDelaySeconds: 60
```

Comment:
the liveness probe delay, to make sense, should be less than readiness probe because  if the container is not alive in 60 seconds it sure wouldn't be ready to accept traffic

Reply:
plus I don't know if it's actually makes a difference to specify same endpoint for readiness & liveness probe, I think if they do the same checks then you can remove readiness probe

Reply:
The readiness will dictate when the pod can start accepting traffic, whereas liveness could trigger a new pod if the existing one is crashing or stuck ... I think they serve different enough purposes that it's ok to have both, maybe tweaking the values independently as we see how it runs?

https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-setting-up-health-checks-with-readiness-and-liveness-probes

Reply:
Thanks for clarifying i had an incorrect understanding for the liveness probe role. Apparently there is a startup probe to wait before running other probes 
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes